Sources/UI/NotificationCenter/NotificationCenterViewController.swift

```diff
@@ -424,3 +346,152 @@ extension NotificationCenterViewController: UIViewControllerTransitioningDelegat
         return SlideRightAnimator()
     }
 }
+
+
+public class NotificationsDataSource: NSObject, UITableViewDataSource {
+    var subscribedTableView: UITableView? = nil
+    
+    public let imageStore: ImageStore
+    
+    let managedObjectContext: NSManagedObjectContext
+    
+    open lazy var fetchRequest: NSFetchRequest<RoverData.Notification> = {
+        let fetchRequest: NSFetchRequest<RoverData.Notification> = RoverData.Notification.fetchRequest()
+        fetchRequest.predicate = self.predicate
+        fetchRequest.sortDescriptors = [
+            NSSortDescriptor.init(key: #keyPath(RoverData.Notification.deliveredAt), ascending: false)
+        ]
+        fetchRequest.fetchBatchSize = 20
+        return fetchRequest
+    }()
+    
+    open lazy var predicate: NSPredicate? = nil
+    
+    open lazy var fetchedResultsController: NSFetchedResultsController<RoverData.Notification> = NSFetchedResultsController(
+        fetchRequest: fetchRequest,
+        managedObjectContext: managedObjectContext,
+        sectionNameKeyPath: nil,
+        cacheName: nil
+    )
+    
+    init(
+        managedObjectContext: NSManagedObjectContext,
+        imageStore: ImageStore
+    ) {
+        self.managedObjectContext = managedObjectContext
+        self.imageStore = imageStore
+
+        super.init()
+        fetchedResultsController.delegate = self
+        do {
+            try fetchedResultsController.performFetch()
+        } catch {
+            os_log("Problem fetching notifications list: %s", log: .ui, error.localizedDescription)
+        }
+    }
+    
+    /**
+     Returns a filtered list of notifications from dataStore.notifications. The default implementation filters out notifications that aren't notification center enabled, are deleted or have expired. This method is called automatically by the NotificationCenterTableView and the results are cached for performance.
+     
+     You can override this method if you wish to modify the rules used to filter notifications. For example if you wish to include expired notifications in the table view and instead show their expired status with a visual indicator.
+     */
+    open var notificationFilterPredicate: NSPredicate {
+        return NSPredicate()
+    }
+    
+    func notificationAt(indexPath: IndexPath) -> RoverData.Notification {
+        return fetchedResultsController.object(at: indexPath) as RoverData.Notification
+    }
+    
+    open func cellReuseIdentifier(at indexPath: IndexPath) -> String {
+        return "notification"
+    }
+    
+    // MARK: UITableViewDataSource
+    
+    open func numberOfSections(in tableView: UITableView) -> Int {
```

Comment:
Since our implementation does not use sections, you can omit this method. The default will return 1 which is what we want. If a developer's implementation does something different they can override this themselves.